whole-tale/gwvolman
# Publish bdbag zip to DataONE

**Problem:**
Our current approach to publishing tales to DataONE transforms the exported BDBag into a DataONE dataset. This is problematic for three reasons: 
1. DataONE doesn't support folders, so any hierarchy is flattened. This presents problems for any tales that have folders and with this release for tales with recorded runs.
2. The resulting published tale cannot be downloaded and run locally, since the downloaded DataONE bag is structured differently.
3. It complicates re-importing tales, since any transformations have to be undone on the roundtrip -- which isn't possible for tales with folders today.

For Zenodo, we publish the zipped bag, which makes download and run-local possible, simplifying tale round-trips. The downside of course is that the contents of the bag can't be inspected without downloading.

**Approach:**
Modify publish to DataONE to publish only the zipped bag. By adding the "Tale" keyword, we are setup to implement import -- ala the Zenodo provider -- in a future release

**To Test:**
* Configure Settings > DataONE > cn-stage.test.dataone.org
* Create a tale with subfolders or import https://girder.local.wholetale.org/api/v1/integration/zenodo?doi=10.5072%2Fzenodo.1028354&resource_server=sandbox.zenodo.org
* Publish to DataONE dev 
* Open dataset on dev.nceas.ucsb.edu
* Confirm `<version_id>.zip` is published, abstract matches description, and keyword "Tale" is present
* Create a new version or recorded run
* Re-publish and confirm prior version on D1 is obsoleted by new version.
* Optionally download, validate bag, and run locally.
